### DISCUSSION TITLE
Sprint 3 Discussion

### DISCUSSION THREAD

**Discussion by Nitya-vaishnavi** [BHAVANA1497/GatorDesk#76] [Ideas]
Discussion board for Sprint 3 so that the entire team is on the same page

**Comment by Nitya-vaishnavi** [comment]
Guys, as we know, me (@Nitya-vaishnavi) and @akhil-maddukuri have worked on the backend code for Session Management in the last sprint. We are planning to test these sessions for each API call for a range of users for this sprint. What are your inputs on this? Any suggestions?

**Reply by akhil-maddukuri** [reply]
Yeah, we can do that. In addition to that, we can also start working on the lost and found feature.

**Reply by Nitya-vaishnavi** [reply]
Also, Ccode from the front-end must be included to redirect to login page when a non-user is trying to access a user-exclusive page (like Announcements or Lost And Found etc). We need this to test sessions. I'll assign this to @HarshithaMyadam and @BHAVANA1497 to include code from the front-end.

**Reply by akhil-maddukuri** [reply]
Yes, it should prompt to login/signup when the user is not already logged in. The front-end team can alter the routing to implement this.

**Comment by Nitya-vaishnavi** [comment]
@BHAVANA1497 @akhil-maddukuri @HarshithaMyadam How do we plan to proceed with Lost and Found functionality? Should only the admin be able to upload the lost and found items or Residents can upload too?

**Reply by akhil-maddukuri** [reply]
Anyone can find a lost item whether he/she is a resident or not, Non-residents cannot log in to the system to post an announcement, right? How do we handle such cases?

**Reply by Nitya-vaishnavi** [reply]
@akhil-maddukuri As we discussed with the rest of the team (@BHAVANA1497 and @HarshithaMyadam ), let us centralise this lost and found process with the help of an 'admin'

**Reply by Nitya-vaishnavi** [reply]
-> Be it a resident or a non-resident, anyone who finds a lost item can upload an image and hand it over to an admin. The admin can exclusively see a list of all the lost items in the database. 
-> Any person who has lost an item in the premises of the Housing complex can raise a lost and found request to the admin. the admin will go through the DB personally to find the lost item for the user. Issues going through the admin instead of a public portal will enhance security and prevent false claims of the lost items.

**Comment by akhil-maddukuri** [comment]
@Nitya-vaishnavi I've created an admin model and i'v also created login endpoint for admin. Please write unit tests to check if it is working fine.

**Reply by Nitya-vaishnavi** [reply]
@akhil-maddukuri Sure!

**Reply by Nitya-vaishnavi** [reply]
@akhil-maddukuri I believe we also need to come up with functionalities like - Only admin can add another admin, only admin can delete another admin etc to make this work. Else any user will be able to operate the lost and found portal.

**Reply by akhil-maddukuri** [reply]
@Nitya-vaishnavi yes, only the admin will be able to see all the lost and found posts. Each individual user will be able to see only his/her posts.

**Comment by Nitya-vaishnavi** [comment]
@akhil-maddukuri I'm facing an issue with the admin table, hence the APIs that I have created are returning the wrong response with an error. Could you please look into it and try to resolve it?

Attached is the screenshot below:

<img width="1071" alt="image" src="https://user-images.githubusercontent.com/43131417/161341478-bc8bd87d-cb08-4514-a78f-f4860887ea4d.png">

**Reply by akhil-maddukuri** [reply]
yeah, I'll look into this

**Reply by akhil-maddukuri** [reply]
@Nitya-vaishnavi , I've resolved it. Now you can add an admin
<img width="1015" alt="image" src="https://user-images.githubusercontent.com/46106443/161343285-2129707c-3942-4916-a214-f2d5fa6446b4.png">
